packages/rest/src/providers/reject.provider.ts

```diff
@@ -11,6 +11,12 @@ import {RestBindings} from '../keys';
 
 const debug = require('debug')('loopback:rest:reject');
 
+// TODO(bajtos) Make this mapping configurable at RestServer level,
+// allow apps and extensions to contribute additional mappings.
+const codeToStatusCodeMapping: {[key: string]: number} = {
+  MODEL_NOT_FOUND: 404,
```

Comment:
This is out custom error code name, right? We'll have to document this and other possible error types. Eg:

1. Database serve not reachable
2. Authentication failure
3. Authorization failure
4. Runtime errors

Reply:
Yes, this is our custom error code. We are using the same `code` property as Node.js core does, as a result we "inherit" all error codes from Node.js core too - see https://nodejs.org/api/errors.html#errors_node_js_error_codes and https://nodejs.org/api/errors.html#errors_common_system_errors

> Database serve not reachable

I think this will typically manifests as `ECONNREFUSED`,  `ECONNRESET` or `ETIMEDOUT`.

> Authentication failure
> Authorization failure
> Runtime errors

Agreed. See existing auth error codes in LB 3.x - e.g. [common/models/user.js](https://github.com/strongloop/loopback/blob/master/common/models/user.js#L245)